handler/oauth2/introspector.go

```diff
@@ -36,32 +36,32 @@ type CoreValidator struct {
 	DisableRefreshTokenValidation bool
 }
 
-func (c *CoreValidator) IntrospectToken(ctx context.Context, token string, tokenType fosite.TokenType, accessRequest fosite.AccessRequester, scopes []string) (fosite.TokenType, error) {
+func (c *CoreValidator) IntrospectToken(ctx context.Context, token string, tokenUse fosite.TokenUse, accessRequest fosite.AccessRequester, scopes []string) (fosite.TokenUse, fosite.AccessTokenType, error) {
```

Comment:
Given that `fosite.AccessTokenType` is `bearer` always, we don't have to change backwards compatiilbity here with an additional parameter! :)

Reply:
Agree!